packages/graphile-utils/__tests__/ExtendSchemaPlugin-pg.test.js

```diff
@@ -718,3 +718,116 @@ it("allows adding a single table entry to a nested type", async () => {
     pgClient.release();
   }
 });
+
+it("allows to retrieve a single scalar value", async () => {
+  const schema = await createPostGraphileSchema(pgPool, ["graphile_utils"], {
+    disableDefaultMutations: true,
+    appendPlugins: [
+      makeExtendSchemaPlugin(build => {
+        const { pgSql: sql } = build;
+        return {
+          typeDefs: gql`
+            extend type User {
+              myCustomScalar: Int! @pgQuery(
+                fragment: ${embed(sql.fragment`(SELECT 100)`)}
+              )
+              myCustomScalarWithFunction: Int! @pgQuery(
+                fragment: ${embed(queryBuilder => sql.fragment`(SELECT 101)`)}
+              )
+              myCustomScalarWithFunctionAndArgument(test: Int!): Int! @pgQuery(
+                fragment: ${embed(
+                  (queryBuilder, args) =>
+                    sql.fragment`(SELECT ${sql.value(args.test)}::integer)`
+                )}
+              )
+            }
+          `,
+        };
+      }),
+    ],
+  });
+  const printedSchema = printSchema(schema);
+  expect(printedSchema).toMatchSnapshot();
+  const pgClient = await pgPool.connect();
+  try {
+    const { data, errors } = await graphql(
+      schema,
+      `
+        query {
+          user: userById(id: 1) {
+            id
+            name
+            myCustomScalar
+            myCustomScalarWithFunction
+            myCustomScalarWithFunctionAndArgument(test: 102)
+          }
+        }
+      `,
+      null,
+      { pgClient },
+      {}
+    );
+    expect(errors).toBeFalsy();
+    expect(data).toBeTruthy();
+    expect(data.user).toBeTruthy();
+    expect(data.user.myCustomScalar).toBe(100);
+    expect(data.user.myCustomScalarWithFunction).toBe(101);
+    expect(data.user.myCustomScalarWithFunctionAndArgument).toBe(102);
+  } finally {
+    pgClient.release();
+  }
+});
+
+it("allows to retrieve array scalar values", async () => {
+  const schema = await createPostGraphileSchema(pgPool, ["graphile_utils"], {
+    disableDefaultMutations: true,
+    appendPlugins: [
+      makeExtendSchemaPlugin(build => {
+        const { pgSql: sql } = build;
+        return {
+          typeDefs: gql`
+            extend type User {
+              myCustomArrayOfScalars: [String!]! @pgQuery(
+                fragment: ${embed(
+                  sql.fragment`array(SELECT name from ${sql.fragment`graphile_utils.pets`})`
```

Comment:
This is equivalent to:

```suggestion
                  sql.fragment`array(SELECT name from graphile_utils.pets)`
```

I've never used the [array constructor](https://www.postgresql.org/docs/12/sql-expressions.html#SQL-SYNTAX-ARRAY-CONSTRUCTORS) in this way; I've always done `select array_agg(name) from ...` instead. It seems that this might actually be [faster because it's simpler](https://dba.stackexchange.com/questions/159710/why-is-array-agg-slower-than-the-non-aggregate-array-constructor) so I should definitely experiment with its usage in PostGraphile.

Reply:
Filed: https://github.com/graphile/graphile-engine/issues/543